src/ScottPlot5/ScottPlot5/IPlottable.cs

```diff
@@ -31,4 +31,5 @@ public interface IPlottable
     /// By default the surface is already clipped to the data area, but this can be cleared inside the plottable.
     /// </summary>
     void Render(SKSurface surface);
+    IList<LegendItem> GetLegendItems();
```

Comment:
I recognize that this function mimics the ScottPlot4 design, but I wonder if there's merit in making it a property?

```cs
IList<LegendItem> LegendItems { get; }
```

I guess most plottables would still need to create the legend items every time it's called (because instantiating the list once and keeping things like colors in sync would be hard), but the API may read a little smoother this way.

Interested in your thoughts

Reply:
I like this, I wonder if it could be turned into an `IEnumerable<LegendItems>` as well, as I waffled on which interface to use for a bit. I used `IEnumerable` in the `Legend` plottable because although it eats the (negligible) performance cost of reifying the enumerable it lets a legend react to its source IEnumerable changing. This means that the default legend will automatically update if/when new plottables are added.

I think caching the legend items is fine with either API, but I frankly wouldn't bother unless somehow there's a clear need. I'd actually go the opposite and turn them all into `IEnumerable`s that aren't turned into concrete lists unless necessary.

Reply:
I turned them all into `IEnumerable<LegendItem> LegendItems { get; }`